### PR TITLE
feat: implement monetization

### PR DESCRIPTION
**Pull request by plun1331**
## Summary

Implements monetization as documented under https://discord.com/developers/docs/monetization

I have no way of testing this as I do not have a monetization-enabled bot (and cannot make one).

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by Nzii3 — APPROVED** [review]

File: discord/enums.py [hunk 1]
```diff
@@ -952,6 +953,19 @@ class ReactionType(Enum):
...
+    subscription = 5
+    subscription_group = 6
+
+
+class EntitlementType(Enum):
+    """The entitlement type"""
+
+    application_subscription = 8
+
+
```
Comment (Lulalaby) [inline discord/enums.py:968]: might miss the enum for owner type
```cs
public enum EntitlementOwnerType
{
	/// <summary>
	/// Owner id is a guild.
	/// </summary>
	GuildSubscription = 1,

	/// <summary>
	/// Owner id is an user.
	/// </summary>
	UserSubscription = 2
}
```

File: discord/guild.py [hunk 2]
```diff
@@ -3842,3 +3843,26 @@ async def delete_auto_moderation_rule(
...
+
+        Returns
+        -------
+        :class:`Entitlement`
+            The created entitlement.
+        """
+        payload = {
+            "sku_id": sku.id,
+            "owner_id": self.id,
+            "owner_type": 1,
```
Comment (Lulalaby) [inline discord/guild.py:3865]: use ownertype enum

File: discord/monetization.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        )
+
+    def __str__(self) -> str:
+        return self.name
+
+    def __eq__(self, other: object) -> bool:
+        return isinstance(other, self.__class__) and other.id == self.id
+
+
+class Entitlement(Hashable):
```
Comment (Lulalaby) [inline discord/monetization.py:100]: it's missing `guild_id`

File: discord/user.py [hunk 4]
```diff
@@ -578,3 +579,26 @@ async def create_dm(self) -> DMChannel:
...
+
+        Returns
+        -------
+        :class:`Entitlement`
+            The created entitlement.
+        """
+        payload = {
+            "sku_id": sku.id,
+            "owner_id": self.id,
+            "owner_type": 2,
```
Comment (Lulalaby) [inline discord/user.py:601]: use ownertype enum

**Review by Lulalaby — CHANGES_REQUESTED** [review]

**Comment by Lulalaby** [comment]
+ merge conflicts

**Review by Lulalaby — DISMISSED** [review]

**Comment by Lulalaby** [comment]
image

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**